bson/unmarshal_value_test.go

```diff
@@ -110,13 +56,11 @@ func BenchmarkSliceCodecUnmarshal(b *testing.B) {
 			bytes:    bsoncore.AppendString(nil, strings.Repeat("t", 4096)),
 		},
 	}
-	reg := NewRegistry()
```

Comment:
Suggest using `NewDecoder` with `SetRegistry` to preserve the original intention of this benchmark.